### ISSUE TITLE
[MNG-3425] Required parameters of self written Mojo configured in execution part of POM are ignored and causing error

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#3823]
**[Halil-Cem Gürsoy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=che---)** opened **[MNG-3425](https://issues.apache.org/jira/browse/MNG-3425?redirect=false)** and commented

Required parameters configured in execution part of POM are ignored if:

the parameter ist set as required AND
(no default value is set OR
default-Value is set to "") AND
configuration of parameter values are done in the execution element.

Example:

Parameter definition in Mojo:

```
/** 
 * @parameter  default-value="ATTENTION DEFAULT"
 * @required
         */	
private File dummy1;
```

and snip from POM:

```xml
<plugins>
  <plugin>
    <groupId>myownartifact.plugin</groupId>
    <artifactId>maven-myownartifact-plugin</artifactId>
    <extensions>true</extensions>
    <executions>
      <execution>
         <id>extract</id>
         <configuration>
          <dummy1>c:/my files/test/dummy1</dummy1>
         </configuration>
         <goals>
           <goal>extract</goal>
         </goals>
      </execution>
    </executions>
  </plugin>     
</plugins>
```

runs fine.

If I change the definition of the parameter to

```
/** 
 * @parameter 
 * @required
         */	
private File dummy1;
```

or

```
* @parameter default-value=""
```

then I got an error:

[0] inside the definition for plugin: 'maven-myownartifact-plugin'specify the following:

\<configuration>
...
\<dummy2>VALUE\</dummy2>
\</configuration>.

In my opinion this is a bug because with `@required` I have already tagged this parameter as mandatory. So I'm already forced to configure it.

BTW, if I configure the parameters outside the execution element evrything works fine as expected!

More info:
- the plugin is bind to the package phase

If a runable example is needed  it's possible to provide one.

---

**Issue Links:**
- [MNG-3401](https://issues.apache.org/jira/browse/MNG-3401) Plugin parameters must be specified outside an execution block when they are invoked from the command line
 (_**"duplicates"**_)

4 votes, 2 watchers

**Comment by jira-importer** [comment]
**[Markku Saarela](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ivalo)** commented

This is very bad bug because I must make my own versions of plugins witch I need to run more than once in same execution with different configuration. One good example is exec-maven-plugin.

**Comment by jira-importer** [comment]
**[Dan Di Spaltro](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dan)** commented

Agreed.  I've had to wrap our mvn execution in a shell script to work around this issue...

**Comment by jira-importer** [comment]
**[Brian E. Fox (imported)](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brianf@infinity.nu)** commented

This is a duplicate.

**Comment by jira-importer** [comment]
**[Brian E. Fox (imported)](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brianf@infinity.nu)** commented

This issue is a duplicate of something that likely cannot be fixed in 2.0.x
